### DISCUSSION TITLE
Where can i find sensor codes for Tasmota-Scripts

### DISCUSSION THREAD

**Discussion by SHU-red** [arendst/Tasmota#18870] [Q&A] [answered]
I am writing a little automation with Tasmota (12.5.0.x) Scripts, which is working so far

I am switching a relay by using
```
spin(5 1)
```
which sets GPIO5 to TRUE

GPIO5 is my Relay 1, configured in the WebUI

Now i want to use
```
pn[x]
````
to read on which GPIO the relay 1 is defined where x is the code for the sensor

I could not find a code list somewhere

Using
```
>B
; list gpio pin definitions
for cnt 0 16 1
tmp=pd[cnt]
print %cnt% = %tmp%
next
```
from documentation returns
```
12:53:46.009 0.00 = 640.00
12:53:46.012 1.00 = 999.00
12:53:46.014 2.00 = 608.00
12:53:46.016 3.00 = 999.00
12:53:46.018 4.00 = 999.00
12:53:46.020 5.00 = 224.00
12:53:46.022 6.00 = 999.00
12:53:46.024 7.00 = 999.00
12:53:46.026 8.00 = 999.00
12:53:46.028 9.00 = 999.00
12:53:46.030 10.00 = 999.00
12:53:46.032 11.00 = 999.00
12:53:46.034 12.00 = 999.00
12:53:46.036 13.00 = 999.00
12:53:46.039 14.00 = 999.00
12:53:46.041 15.00 = 999.00
12:53:46.043 16.00 = 999.00
```
so i guess that relay 1 sensor code is 224

but using
```
pn[224]
```
returns `-1`

What am i not getting right or oversee in the documentation?

**Comment by barbudor** [comment]
Components are documented [here](https://tasmota.github.io/docs/Components/#gpio-overview) and 224 is correctly the code for Realy_1

Now `pn[224]` should definitively return `5` as per the doc.
May be @gemu2015 can advise ?

**Reply by gemu2015** [reply, marked as answer]
codes are now shifted in newer versions of tasmota:  try shift right 5 times
pn[224>>5] should return 5 in your case

**Reply by SHU-red** [reply]
Shifting >>5 seems to return 5

Thank you
